### PR TITLE
feat(intrinsic-function): Adding new intrinsic function for retrieving JSON data

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

-  Add a new `getJson` intrinsic function
- if no path is provided, the whole document is returned as JSON and can be manipulated using feel
- otherwise, only the selected part will be returned

### CODE REVIEW

File: connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  @IntrinsicFunction(name = "getJson")
+  public Object execute(Document document, @Nullable String feelExpression) {
+    try {
+      Object json = objectMapper.readValue(document.asByteArray(), Object.class);
+      if (feelExpression == null || feelExpression.isBlank()) {
+        return json;
+      }
+      // FEEL expects variables as a context map, so wrap the JSON as a variable
+      return feelEngine.evaluate(feelExpression, json);
```
Comment (johnBgood) [inline connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java:40]: @chillleader this would introduce a circular dependency, any thoughts?

Comment (sbuettner) [reply connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java:40]: @johnBgood Only if internal object mapper of the feel engine would have the intrinsic function deserializer I guess. This FEEL engine

Comment (sbuettner) [reply connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java:40]: @johnBgood Maybe the AI can provide a test for it 😄

Comment (johnBgood) [reply connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java:40]: I meant there is a circular deps, we need FeelEngineWrapper which is in `connector-feel-wrapper` which depends on this module

Comment (chillleader) [reply connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java]: Yes, I tested it locally. We should backport to 8.7

**Comment by johnBgood** [comment]
docs: https://github.com/camunda/camunda-docs/pull/5934

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*

File: connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java (hunk 1 above)
Comment (sbuettner) [reply connector-sdk/document/src/main/java/io/camunda/intrinsic/functions/GetJsonFunction.java]: I think it would be great to backport it.

**Review by sbuettner — APPROVED** [review]
Great collaboration @johnBgood & @chillleader where the major work was not done by AI 😄
